runtime/caml/address_class.h

```diff
@@ -72,20 +72,25 @@
 
 #define Is_in_heap(a) (Classify_addr(a) & In_heap)
 
-#define Is_in_heap_or_young(a) (Classify_addr(a) & (In_heap | In_young))
-
 #ifdef NO_NAKED_POINTERS
+
+#define Is_in_heap_or_young(a) 1
 #define Is_in_value_area(a) 1
+
 #else
+
+#define Is_in_heap_or_young(a) (Classify_addr(a) & (In_heap | In_young))
+
 #define Is_in_value_area(a)                                     \
   (Classify_addr(a) & (In_heap | In_young | In_static_data))
-#endif
 
 #define Is_in_code_area(pc) \
  (    ((char *)(pc) >= caml_code_area_start && \
        (char *)(pc) <= caml_code_area_end)     \
    || (Classify_addr(pc) & In_code_area) )
 
+#endif
+
 #define Is_in_static_data(a) (Classify_addr(a) & In_static_data)
```

Comment:
Perhaps `Is_in_static_data` could also be made part of the `#else`.

Reply:
`Is_in_static_data` will be changed as soon as #9678 is merged.  There is one test that uses this macro and needs to be turned off in no-naked-pointers mode, and this work is done as part of #9678.